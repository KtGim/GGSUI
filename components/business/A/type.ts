interface AP {
    height: number;
}


export {
    AP
}