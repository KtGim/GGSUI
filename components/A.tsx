import React from 'react';

const A = () => {
    return (
        <div>
            AAAA
        </div>
    )
};

export default A;