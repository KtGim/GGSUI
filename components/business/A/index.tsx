import React, { useState } from 'react';
import { Button } from 'antd';
import { AP } from './type';
interface Props extends AP {
    name?: string;
    age: number;
}

type AType = "A";

const A = (props: Props) => {
    const [a, setA] = useState(0);
    const aa: AType = 'A';

    return (
        <div className="a-component">
            {a}
            <Button onClick={() => {
                setA(a + 1);
            }}>Button</Button>
        </div>
    )
};

export default A;