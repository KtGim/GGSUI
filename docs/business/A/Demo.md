# heading

## 简介

### 例子

``` demo
<A />
```  

```tsx
  import React from 'react';

  const A = () => {
      return (
          <div>
              AAAA
          </div>
      )
  };

  export default A;
```

### API

| 成员 | 说明 | 类型 | 默认值 | 版本 |
| --- | --- | --- | --- | --- |
| affix | 固定模式 | boolean | true |  |
| bounds | 锚点区域边界 | number | 5 |  |
| getContainer | 指定滚动的容器 | () => HTMLElement | () => window |  |
| getCurrentAnchor | 自定义高亮的锚点 | () => string | - |  |
| offsetTop | 距离窗口顶部达到指定偏移量后触发 | number |  |  |
| showInkInFixed | `affix={false}` 时是否显示小圆点 | boolean | false |  |
| targetOffset | 锚点滚动偏移量，默认与 offsetTop 相同 | number | - |  |
| onChange | 监听锚点链接改变 | (currentActiveLink: string) => void | - |  |
| onClick | `click` 事件的 handler | function(e: Event, link: Object) | - |  |
