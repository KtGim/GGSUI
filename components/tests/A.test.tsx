import React from 'react';
import renderer from 'react-test-renderer';

import A from '../business/A/index';
it('renders correctly', () => {
    const tree = renderer
      .create(<A />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});
test('A is running', () => {
  const component = renderer.create(
    <A />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})