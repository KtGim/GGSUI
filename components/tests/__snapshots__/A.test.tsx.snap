// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A is running 1`] = `
<div
  className="a-component"
>
  0
  <button
    className="ant-btn ant-btn-default"
    onClick={[Function]}
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="a-component"
>
  0
  <button
    className="ant-btn ant-btn-default"
    onClick={[Function]}
    type="button"
  >
    <span>
      Button
    </span>
  </button>
</div>
`;
